UPtables/app/views/unavailabilities/index.html.haml

```diff
@@ -0,0 +1,20 @@
+%h1 Unavailabilities for #{@teacher.full_name}
+= form_tag(set_teacher_unavailabilities_path, :method => :post) do
+  %table
+    %thead
+      %tr
+        %th
+        - @weekdays.each do |day|
+          %th= day.name          
+    %tbody
+      - @timeframes.each do |frame|
+        %tr
+          %td= frame.interval 
+          - @weekdays.each do |day|
+            %td= check_box_tag "date[#{day.id}][#{frame.id}]", "checked", @unavailabilities.any?{|u| u.weekday.id == day.id and u.timeframe.id == frame.id}
+            
```

Comment:
Use logical operators ```||``` and ```&&``` in if clauses and ```and``` ```or``` for control flow. See [here](http://devblog.avdi.org/2010/08/02/using-and-and-or-in-ruby/)